service/awsconfig/v10/resource/ebsvolume/delete.go

```diff
@@ -16,21 +14,35 @@ func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange inte
 		return microerror.Mask(err)
 	}
 
-	if deleteInput != nil && len(deleteInput.VolumeIDs) > 0 {
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleting %d ebs volumes", len(deleteInput.VolumeIDs)))
+	if deleteInput != nil && len(deleteInput.Volumes) > 0 {
+		r.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("deleting %d ebs volumes", len(deleteInput.Volumes)))
 
-		for _, volID := range deleteInput.VolumeIDs {
-			_, err := r.clients.EC2.DeleteVolume(&ec2.DeleteVolumeInput{
-				VolumeId: aws.String(volID),
-			})
+		// First detach any attached volumes without forcing but shutdown the
+		// instances.
+		for _, vol := range deleteInput.Volumes {
+			for _, a := range vol.Attachments {
+				r.service.DetachVolume(ctx, vol.VolumeID, a, false, true)
+			}
+		}
+
+		// Now force detach so the volumes can be deleted cleanly.
+		for _, vol := range deleteInput.Volumes {
+			for _, a := range vol.Attachments {
+				r.service.DetachVolume(ctx, vol.VolumeID, a, true, false)
```

Comment:
Can we maybe improve the comments or have variables for the booleans? I just read the interface above and already forgot which parameter stands for which.

Reply:
Jeje I originally had it as vars and changed it. I've put it back as yes I think its more explicit when read.